src/DataWrangling/ECCO/ECCODarwin.jl

```diff
@@ -0,0 +1,491 @@
+    username = get(ENV, "ECCO_USERNAME", nothing)
+    password = get(ENV, "ECCO_PASSWORD", nothing)
+    dir = metadata.dir
+    
+    # Create a temporary directory to store the .netrc file
+    # The directory will be deleted after the download is complete
+    @root mktempdir(dir) do tmp
+
+        # Write down the username and password in a .netrc file
+        downloader = netrc_downloader(username, password, "ecco.jpl.nasa.gov", tmp)
+
+        asyncmap(metadata, ntasks=10) do metadatum # Distribute the download among tasks
+
+            fileurl  = metadata_url(url, metadatum) 
+            filepath = metadata_path(metadatum)
+
+            if !isfile(filepath)
+                instructions_msg = "\n See ClimaOcean.jl/src/ECCO/README.md for instructions."
+                if isnothing(username)
+                    msg = "Could not find the ECCO_PASSWORD environment variable. \
+                           See ClimaOcean.jl/src/ECCO/README.md for instructions on obtaining \
+                           and setting your ECCO_USERNAME and ECCO_PASSWORD." * instructions_msg
+                    throw(ArgumentError(msg))
+                elseif isnothing(password)
+                    msg = "Could not find the ECCO_PASSWORD environment variable. \
+                           See ClimaOcean.jl/src/ECCO/README.md for instructions on obtaining \
+                           and setting your ECCO_USERNAME and ECCO_PASSWORD." * instructions_msg
+                    throw(ArgumentError(msg))
+                end
+                # Get the data file
+                Downloads.download(fileurl, filepath; downloader, progress=download_progress)
+                # Get the meta file
+                Downloads.download(replace(fileurl, "data" => "meta"), 
+                                   replace(filepath, "data" => "meta"); 
+                                   downloader, 
+                                   progress=download_progress)
+            end
+        end
+    end
+    
+    return nothing
+end
+
+ECCODarwinMeshGrid(::ECCO4DarwinMonthly)   = GridSpec("LatLonCap", MeshArrays.GRID_LLC90)
+ECCODarwinMeshGrid(::ECCO270DarwinMonthly) = gcmgrid("./", "LatLonCap", 5, [(270, 810), (270, 810), (270, 270), (810, 270), (810, 270)], (270, 3510), Float64, read, write)
+
+"""
+    ECCODarwinModelMeta(metafile)
+
+Read ECCODarwin metadata file and return as a NamedTuple
+
+"""
+function ECCODarwinModelMeta(fileName)
+    meta = read(fileName,String)
+    meta = split(meta,";\n")
+    meta = meta[isempty.(meta).==false]
+    meta = replace.(meta,Ref(",\n"=>";"))
+    meta = replace.(meta,Ref("\n"=>""))
+    meta = replace.(meta,Ref("}"=>"]"))
+    meta = replace.(meta,Ref("{"=>"["))
+    meta = replace.(meta,Ref("'"=>"\""))
+    meta = replace.(meta,Ref(";]"=>"]"))
+    meta = replace.(meta,Ref(","=>" "))
+
+    meta = split.(meta,"=")
+    meta = [[replace(x[1]," "=>"") x[2]] for x in meta]
```

Comment:
```suggestion
    meta = [[replace(x[1], " " => ""), x[2]] for x in meta]
```

Note that a tuple may be more appropriate than an array in some of these cases. Arrays should be used when either the entries or length of the array need to change. For small immutable collections like `(1, 2)`, a tuple is more appropriate, because on inspection of the code you know that the entries and length of the collection will not change.